src/app/services/data/node.service.ts

```diff
@@ -354,7 +376,7 @@ export class NodeService implements OnDestroy {
     }
   }
 
-  undockTransition(nodeId: number, transitionId: number, enforceUpdate = true) {
+  undockTransition(nodeId: number, transitionId: number, enforceUpdate = true): TrainrunSection {
```

Comment:
```suggestion
  undockTransition(nodeId: number, transitionId: number, enforceUpdate = true): TrainrunSection | undefined {
```

Otherwise it might cause typing issues difficult to detect